File: ios/LedgerFlow/Models/UserModels.swift
import Foundation
import FirebaseAuth
import FirebaseFirestore

class User: Codable {
    let id: String
    let email: String
    let name: String
    let accountType: String // "individual" or "company"
    let currency: String
    let createdAt: Date
    let updatedAt: Date
    let emailVerified: Bool
    
    init(
        id: String = UUID().uuidString,
        email: String,
        name: String,
        accountType: String,
        currency: String,
        createdAt: Date = Date(),
        updatedAt: Date = Date(),
        emailVerified: Bool = false
    ) {
        self.id = id
        self.email = email
        self.name = name
        self.accountType = accountType
        self.currency = currency
        self.createdAt = createdAt
        self.updatedAt = updatedAt
        self.emailVerified = emailVerified
    }
    
    func toDictionary() -> [String: Any] {
        return [
            "id": id,
            "email": email,
            "name": name,
            "accountType": accountType,
            "currency": currency,
            "createdAt": createdAt,
            "updatedAt": updatedAt,
            "emailVerified": emailVerified
        ]
    }
}

class CompanyUser: Codable {
    let id: String
    let companyId: String
    let email: String
    let name: String
    let role: String // "admin", "manager", "accountant", "viewer"
    let createdAt: Date
    let isActive: Bool
    
    init(
        id: String = UUID().uuidString,
        companyId: String,
        email: String,
        name: String,
        role: String,
        createdAt: Date = Date(),
        isActive: Bool = true
    ) {
        self.id = id
        self.companyId = companyId
        self.email = email
        self.name = name
        self.role = role
        self.createdAt = createdAt
        self.isActive = isActive
    }
    
    func toDictionary() -> [String: Any] {
        return [
            "id": id,
            "companyId": companyId,
            "email": email,
            "name": name,
            "role": role,
            "createdAt": createdAt,
            "isActive": isActive
        ]
    }
}

File: android/app/src/main/kotlin/com/ledgerflow/models/UserModels.kt
package com.ledgerflow.models

import java.io.Serializable
import java.util.*

data class User(
    val id: String = UUID.randomUUID().toString(),
    val email: String,
    val name: String,
    val accountType: String, // "individual" or "company"
    val currency: String,
    val createdAt: Date = Date(),
    val updatedAt: Date = Date(),
    val emailVerified: Boolean = false
) : Serializable {
    fun toMap(): Map<String, Any> {
        return mapOf(
            "id" to id,
            "email" to email,
            "name" to name,
            "accountType" to accountType,
            "currency" to currency,
            "createdAt" to createdAt,
            "updatedAt" to updatedAt,
            "emailVerified" to emailVerified
        )
    }
}

data class CompanyUser(
    val id: String = UUID.randomUUID().toString(),
    val companyId: String,
    val email: String,
    val name: String,
    val role: String, // "admin", "manager", "accountant", "viewer"
    val createdAt: Date = Date(),
    val isActive: Boolean = true
) : Serializable {
    fun toMap(): Map<String, Any> {
        return mapOf(
            "id" to id,
            "companyId" to companyId,
            "email" to email,
            "name" to name,
            "role" to role,
            "createdAt" to createdAt,
            "isActive" to isActive
        )
    }
}
